pkg/internal/codegen/parse/crd.go

```diff
@@ -120,6 +120,14 @@ func (b *APIs) parseCRDs() {
 						}
 						resource.CRD.Spec.AdditionalPrinterColumns = result
 					}
+					if IsAPIResource(resource.Type) {
+						result, err := parseResourceTag(resource.Type)
+						if err != nil {
+							log.Fatalf("failed to parse resource annotations, error: %v", err.Error())
+						}
+						resource.CRD.ObjectMeta.Name = result.Resource + "." + resource.Group + "." + resource.Domain
+						resource.CRD.Spec.Names.Plural = result.Resource
```

Comment:
this seems like it should be done when we parse the original resource.

Reply:
I have added it after we parse the original resource so that if it will override the existing values (if any).